pkg/translate/translateValue.go

```diff
@@ -96,20 +95,38 @@ func (t *ValueYAMLTranslator) initAPIMapping(vs version.MinorVersion) {
 	}
 }
 
-// initK8SMapping generates the k8s settings mapping for all components based on templates
-func (t *ValueYAMLTranslator) initK8SMapping() {
+// initK8SMapping generates the k8s settings mapping for components that are enabled based on templates
+func (t *ValueYAMLTranslator) initK8SMapping(valueTree map[string]interface{}) {
 	outputMapping := make(map[string]*Translation)
 	for valKey, componentName := range t.ValuesToComponentName {
 		featureName := name.ComponentNameToFeatureName[componentName]
 		if featureName == "" {
 			continue
 		}
+		cnEnabled, err := name.IsComponentEnabledFromValue(valKey, valueTree)
+		if err != nil {
+			log.Errorf("Error checking component:%s enablement, skip k8s mapping", componentName)
+			continue
+		}
+		if !cnEnabled {
+			log.Infof("Component:%s disabled, skip k8s mapping", componentName)
+			continue
+		}
 		for K8SValKey, outPathTmpl := range t.KubernetesMapping {
 			newKey := renderComponentName(K8SValKey, valKey)
 			newVal := renderFeatureComponentPathTemplate(outPathTmpl.outPath, featureName, componentName)
 			outputMapping[newKey] = &Translation{newVal, nil}
 		}
 	}
+
+	// add readiness mapping
+	outputMapping["global.proxy.readinessInitialDelaySeconds"] = &Translation{
```

Comment:
Why are these not part of the translator mappings data?

Reply:
this is actually part of the translator mappings data, but we just did not use template to generate them since there is special path for readiness